pkg/alertmanager/validation.go

```diff
@@ -241,6 +244,15 @@ func validatePushoverConfigs(configs []monitoringv1alpha1.PushoverConfig) error
 	return nil
 }
 
+func validateSnsConfigs(configs []monitoringv1alpha1.SNSConfig) error {
+	for _, config := range configs {
+		if (config.TargetARN == "") != (config.TopicARN == "") != (config.PhoneNumber == "") {
```

Comment:
What would happen here if none of the values are set? By my interpretation this condition will be true, but I'm not sure if that's what we want.

Reply:
It would evaluate to `true != true != true` -> `false != true` -> `true` so it works as expected but I had to think twice :sweat_smile: 
FWIW the test is taken from the upstream Alertmanager code base.

Reply:
Oh wow, totally unexpected but it does seem to work out in the end :smile:

Reply:
Yes, as mentioned this (and some other validations) are taken directly from the alertmanager project. Agree that it is not the easiest code to read so we can change it here if ye prefer?

Reply:
I am fine with leaving it as it is, I learned something new from it :)